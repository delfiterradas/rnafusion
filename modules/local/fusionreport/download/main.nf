process FUSIONREPORT_DOWNLOAD {
    tag 'fusionreport'
    label 'process_medium'

    conda "bioconda::star=2.7.9a"
    container "docker.io/clinicalgenomics/fusion-report:3.1.0"

    output:
    tuple val(meta), path("fusionreport_dbs"), emit: fusionreport_db
    path "versions.yml"                      , emit: versions

    script:
    meta = [id: 'fusionreport_dbs']
    def args = task.ext.args ?: ''
    def args2 = task.ext.args2 ?: ''
    """
    fusion_report download $args ./
    mkdir fusionreport_dbs
    mv *.txt *.log *.db fusionreport_dbs

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        fusion_report: \$(fusion_report --version | sed 's/fusion-report //')
    END_VERSIONS
    """

    stub:
    meta = [id: 'fusionreport_dbs']
    """
    mkdir fusionreport_dbs
    touch fusionreport_dbs/cosmic.db
    touch fusionreport_dbs/fusiongdb2.db
    touch fusionreport_dbs/mitelman.db
    touch fusionreport_dbs/DB-timestamp.txt
    touch fusionreport_dbs/fusion_report.log

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        fusion_report: \$(fusion_report --version | sed 's/fusion-report //')
    END_VERSIONS
    """
}
