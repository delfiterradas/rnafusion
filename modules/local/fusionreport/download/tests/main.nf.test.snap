{
    "Download fusionreport databases": {
        "content": [
            "fusiongdb2.db:md5,e1ac123a744e515d3e5f85b8344d526a",
            "mitelman.db:md5,1363795c97f77c641065ecd9ad0e484a",
            true,
            true,
            [
                "versions.yml:md5,fa5f13c563f431912048c1802b5a0c74"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-12-05T19:27:38.99855171"
    },
    "Create stub files": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "fusion_report_db"
                        },
                        [
                            "DB-timestamp.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "cosmic.db:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "fusion_report.log:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "fusiongdb2.db:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "mitelman.db:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,fa5f13c563f431912048c1802b5a0c74"
                ],
                "fusionreport_ref": [
                    [
                        {
                            "id": "fusion_report_db"
                        },
                        [
                            "DB-timestamp.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "cosmic.db:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "fusion_report.log:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "fusiongdb2.db:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "mitelman.db:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,fa5f13c563f431912048c1802b5a0c74"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-12-10T15:05:22.781845"
    }
}